### ISSUE TITLE
Filters?

### ISSUE THREAD

**Issue by Licenser** [census-instrumentation/opencensus-erlang#4]
Is there an interest in porting the filter implementation from otters?

The filters are built to be extremely performant and have minimal impact when run, they're defined with a [filter language](https://github.com/project-fifo/otters#filtering-language) that resembles erlang itself closely and then compiled to an erlang module. This means they're running as code and taking advantage of Erlangs optimisations when it comes to matching rather as data to compute over and compare.

**Comment by tsloughter** [comment]
Not sure what their plans, if any, are for that functionality. I definitely agree porting over otters implementation would be the best bet. But could need a standard format for the rules on top if that is part of census.

**Comment by Licenser** [comment]
yup I figured there might be so I wanted to ask before jumping the gun :)

**Comment by tsloughter** [comment]
Goldrush is for a similar purpose in lager, right? Do you think there would be any advantage to using goldrush, and to be configured the same as a rule in lager? 

Also, we should broaden this discussion to all of census probably? I can open an issue in https://github.com/census-instrumentation/opencensus-specs

**Comment by tsloughter** [comment]
@Licenser we should definitely have a reporter that uses your filter language I think. I can look at adding that if you want.

**Comment by deadtrickster** [comment]
I'm interested in filters too. Are they supposed to be in reporters? Or in the storage?

**Comment by deadtrickster** [comment]
@tsloughter now I think filters are useful not only for reporters - I'm playing with filtering before storing span.
@Licenser have you touched this?

**Comment by tsloughter** [comment]
@deadtrickster any update on this? Are you using @Licenser 's code from otter? https://github.com/project-fifo/otters#filtering-language

**Comment by deadtrickster** [comment]
well not really, but still compiled to module. Differences:
1. not another language - config goes to sys.config
2. no actions - just true/false. False means span won't be stored. (I don't really know whether actions thing is relevant for opencensus, i.e. the only action is "to store"?)

**Comment by tsloughter** [comment]
Ah yea, good point about actions.

I too likely prefer a sys.config entry since that'll make it easier to configure in Kubernetes deploys where I want to adjust values with env vars.

**Comment by deadtrickster** [comment]
I likely need more time to get this ready. will create pull request
